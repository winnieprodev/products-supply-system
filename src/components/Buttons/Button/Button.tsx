import React from "react";

import * as Styled from "./button.styled";

interface OwnProps {
  fullWidth?: boolean;
  primary?: boolean;
  transparent?: boolean;
  onClick?: () => void;
  disabled?: boolean;
  children: string;
}

const Button: React.FC<OwnProps> = ({
  fullWidth = false,
  primary = false,
  transparent = false,
  disabled = false,
  ...buttonProps
}) => (
  <Styled.Button
    $fullWidth={fullWidth}
    $primary={primary}
    $transparent={transparent}
    disabled={disabled}
    {...buttonProps}
  />
);

export default React.memo(Button);
