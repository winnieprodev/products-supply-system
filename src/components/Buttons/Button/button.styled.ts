import styled, { css } from "styled-components";

import { ButtonCore } from "../buttons.styled";

interface ButtonProps {
  $fullWidth?: boolean;
}

export const Button = styled(ButtonCore)<ButtonProps>`
  border: 0;
  border-radius: 8px;
  padding: 8px 16px;
  margin: 4px;
  font-weight: 700;
  ${({ $fullWidth }) =>
    $fullWidth &&
    css`
      width: 100%;
      margin: 0;
    `}
`;
