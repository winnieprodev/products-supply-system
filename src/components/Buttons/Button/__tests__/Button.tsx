import { render } from "@testing-library/react";

import Button from "../Button";

describe("Button", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot with min. props", () => {
      const { container } = render(<Button>test</Button>);
      expect(container.firstChild).toMatchSnapshot();
    });
    test("it matches the snapshot with fullWidth = true", () => {
      const { container } = render(<Button fullWidth>test</Button>);
      expect(container.firstChild).toMatchSnapshot();
    });
    test("it matches the snapshot with transparent = true", () => {
      const { container } = render(<Button transparent>test</Button>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
