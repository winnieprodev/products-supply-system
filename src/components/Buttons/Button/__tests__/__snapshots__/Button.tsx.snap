// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshot tests it matches the snapshot with fullWidth = true 1`] = `
.c0 {
  outline: none;
  cursor: pointer;
  border: 0 none;
  -webkit-transition: background ease 0.3s;
  transition: background ease 0.3s;
  background-color: #EEEEEE;
  color: #5F5F5F;
}

.c0:hover {
  background-color: #E7E7E7;
}

.c0:active {
  background-color: #F3F3F3;
}

.c1 {
  border: 0;
  border-radius: 8px;
  padding: 8px 16px;
  margin: 4px;
  font-weight: 700;
  width: 100%;
  margin: 0;
}

<button
  class="c0 c1"
>
  test
</button>
`;

exports[`Button Snapshot tests it matches the snapshot with min. props 1`] = `
.c0 {
  outline: none;
  cursor: pointer;
  border: 0 none;
  -webkit-transition: background ease 0.3s;
  transition: background ease 0.3s;
  background-color: #EEEEEE;
  color: #5F5F5F;
}

.c0:hover {
  background-color: #E7E7E7;
}

.c0:active {
  background-color: #F3F3F3;
}

.c1 {
  border: 0;
  border-radius: 8px;
  padding: 8px 16px;
  margin: 4px;
  font-weight: 700;
}

<button
  class="c0 c1"
>
  test
</button>
`;

exports[`Button Snapshot tests it matches the snapshot with transparent = true 1`] = `
.c0 {
  outline: none;
  cursor: pointer;
  border: 0 none;
  -webkit-transition: background ease 0.3s;
  transition: background ease 0.3s;
  background-color: #EEEEEE;
  color: #5F5F5F;
  background-color: transparent;
}

.c0:hover {
  background-color: #E7E7E7;
}

.c0:active {
  background-color: #F3F3F3;
}

.c1 {
  border: 0;
  border-radius: 8px;
  padding: 8px 16px;
  margin: 4px;
  font-weight: 700;
}

<button
  class="c0 c1"
>
  test
</button>
`;
